### PR TITLE
CLN: __all__ for core/index.py

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Review by WillAyd — APPROVED** [review]
lgtm

File: pandas/core/index.py [hunk 1]
```diff
@@ -20,7 +20,30 @@
     ensure_index_from_sequences,
     get_objs_combined_axis,
 )
-from pandas.core.indexes.multi import _sparsify  # noqa:F401
```
Comment (WillAyd) [inline pandas/core/index.py]: Is this actually imported from anywhere else? If not I think could actually be removed

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/core/index.py [hunk 2]
```diff
@@ -20,7 +20,30 @@
...
+    "NumericIndex",
+    "PeriodIndex",
+    "RangeIndex",
+    "TimedeltaIndex",
+    "UInt64Index",
+    "_new_Index",
+    "ensure_index",
+    "ensure_index_from_sequences",
+    "get_objs_combined_axis",
+    "_sparsify",
```
Comment (datapythonista) [inline pandas/core/index.py]: I don't think the private methods `_new_Index` and `_sparsify` should be in `__all__`. I guess they are being used by other modules, which shouldn't be the case.

I'm ok to merge it like this, if that's the case, but if you don't mind researching a bit, seeing where those are used, and adding at least a comment about this, that would be great.

I'd probably move these imports at the end, so they are separated from the rest.

**Comment by MomIsBestFriend** [comment]
this file is not used __anywhere__ in the code base (AT ALL).
It was deprecated in #30193 (About a month ago).

Is it OK to removed it completely?

I don't know if a month is a lot of time or not when it comes to API changes, if it's not a lot of time, then what is a good time span to give users a warn about API change before actually changing it?

**Comment by datapythonista** [comment]
The CI is red, I guess the objects you removed were being imported somewhere.

**Comment by MomIsBestFriend** [comment]
Looks like it failed before it began the tests, closing and reopening

**Comment by datapythonista** [comment]
Sorry, assumed there were genuine test fails. The failure is a correct error now, you need to either remove `_new_Index` if it's not used, or add it to the `__all__` (with a comment if possible).

**Comment by jorisvandenbossche** [comment]
This is a deprecated module that is being changed, that should not be used in pandas. So what is the rationale for this change?

**Comment by MomIsBestFriend** [comment]
> This is a deprecated module that is being changed, that should not be used in pandas. So what is the rationale for this change?

@jorisvandenbossche I just saw the ```# noqa:F401``` comments, and decided to cleaned it up, regardless if the module is deprecated or not.

**Comment by jorisvandenbossche** [comment]
OK, I wouldn't put effort in a deprecated / unused file, but too late I guess :-)

Is there a general issue about such changes?

**Comment by MomIsBestFriend** [comment]
> but too late I guess :-)

That's exactly what I said, when I saw the deprecation message, only after I have made the changes :)

> Is there a general issue about such changes?

There isn't a "master issue" for it, it's something I started to do here and there since https://github.com/pandas-dev/pandas/pull/30828#discussion_r364500688 was pointed out.

**Comment by WillAyd** [comment]
Yea I think since deprecated maybe not even worth touching this module so just closing this one out. But thanks as always @MomIsBestFriend

**Comment by MomIsBestFriend** [comment]
Understandable :)

**Pull request closed without merging**